dexcore/core.go

```diff
@@ -0,0 +1,92 @@
+package dexcore
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/dex"
+	libutils "github.com/crypto-power/cryptopower/libwallet/utils"
+)
+
+type Core interface {
+	Run(ctx context.Context)
+	Ready() <-chan struct{}
+	IsInitialized() bool
+	InitializeClient(pw, seed []byte) error
+	DiscoverAccount(dexAddr string, pass []byte, certI any) (*core.Exchange, bool, error)
+	BondsFeeBuffer(assetID uint32) uint64
+	PostBond(form *core.PostBondForm) (*core.PostBondResult, error)
+	NotificationFeed() <-chan core.Notification
+	Login(pw []byte) error
+}
+
+// New returns a *DEXCore that satisfies Core.
+func New(root, lang string, net libutils.NetworkType) (*DEXCore, error) {
+	dexNet := toDEXNet(string(net))
+	network, err := dex.NetFromString(dexNet)
+	if err != nil {
+		return nil, fmt.Errorf("dex.NetFromString error: %w", err)
+	}
+
+	dbPath := filepath.Join(root, string(net), "dexc.db")
+	cfg := &core.Config{
+		DBPath:             dbPath,
+		Net:                network,
+		Logger:             log,
+		Language:           lang,
+		UnlockCoinsOnLogin: true,
+		NoAutoWalletLock:   true,
+	}
+
+	clientCore, err := core.New(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize dex core: %w", err)
+	}
+
+	dc := &DEXCore{
+		Core:            clientCore,
+		bondBufferCache: make(map[uint32]valStamp),
+	}
+
+	return dc, err
+}
+
+// BondsFeeBuffer is a caching helper for the bonds fee buffer. Values for a
+// given asset are cached for 45 minutes. These values are meant to provide a
+// sensible but well-padded fee buffer for bond transactions now and well into
+// the future, so a long expiry is appropriate.
+func (dc *DEXCore) BondsFeeBuffer(assetID uint32) (feeBuffer uint64) {
```

Comment:
Why do we need a cache. Don't see that in the comment.

Reply:
The reason is the first comment in the method, moved it though.